gabimoreno/src/test/java/soy/gabimoreno/presentation/screen/courses/list/AudioCoursesListViewModelTest.kt

```diff
@@ -88,17 +93,48 @@ class AudioCoursesListViewModelTest {
         }
 
         viewModel = AudioCoursesListViewModel(
-            getCoursesUseCase,
+            getCoursesUseCase = getCoursesUseCase,
             getShouldIReloadAudioCoursesUseCase = getShouldIReloadAudioCoursesUseCase,
             setShouldIReloadAudioCoursesUseCase = setShouldIReloadAudioCoursesUseCase,
-            testDispatcher
+            refreshBearerTokenUseCase = refreshBearerTokenUseCase,
+            dispatcher = testDispatcher
         )
+
         advanceUntilIdle()
 
         eventChannel.receive() shouldBeEqualTo AudioCoursesListEvent.ShowTokenExpiredError
         job.cancel()
     }
 
+    @Test
+    fun `GIVEN token expired THEN token refreshed and courses loaded`() = runTest {
+        val dummyCourses = buildAudioCourses()
+
+        coEvery { getCoursesUseCase(any()) } returnsMany listOf(
+            TokenExpiredException().left(),
+            dummyCourses.right()
+        )
+        coEvery { refreshBearerTokenUseCase() } returns Unit.right()
+
+        viewModel = AudioCoursesListViewModel(
+            getCoursesUseCase = getCoursesUseCase,
+            getShouldIReloadAudioCoursesUseCase = getShouldIReloadAudioCoursesUseCase,
+            setShouldIReloadAudioCoursesUseCase = setShouldIReloadAudioCoursesUseCase,
+            refreshBearerTokenUseCase = refreshBearerTokenUseCase,
+            dispatcher = testDispatcher
+        )
+
+        advanceUntilIdle()
+
+        viewModel.state.audiocourses shouldBeEqualTo dummyCourses
+        coVerifyOnce {
+            refreshBearerTokenUseCase()
+        }
+        coVerify(exactly = 2) {
+            getCoursesUseCase(any())
```

Comment:
Could we be mroe explicit than `any()`?
`any()` is often useful for `verifyNever` stuff.